### PR TITLE
New package: date

### PR DESCRIPTION
**Pull request by gmgunter**

### CODE REVIEW

File: var/spack/repos/builtin/packages/date/package.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            args.append('-DBUILD_SHARED_LIBS=ON')
+
+        if spec.variants['tz'].value:
+            args.append('-DBUILD_TZ_LIB=ON')
+
+        tzdb = spec.variants['tzdb'].value
+        if tzdb == 'system':
+            args.append('-DUSE_SYSTEM_TZ_DB=ON')
+        elif tzdb == 'manual':
+            args.append('-DMANUAL_TZ_DB=ON')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/date/package.py]: Can you explicitly disable these options in an else clause?

File: var/spack/repos/builtin/packages/date/package.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            spec.variants['cxxstd'].value))
+        # Require standard at configure time to guarantee the
+        # compiler supports the selected standard.
+        args.append('-DCMAKE_CXX_STANDARD_REQUIRED=ON')
+
+        if spec.variants['shared'].value:
+            args.append('-DBUILD_SHARED_LIBS=ON')
+
+        if spec.variants['tz'].value:
+            args.append('-DBUILD_TZ_LIB=ON')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/date/package.py]: Same here

File: var/spack/repos/builtin/packages/date/package.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        args = []
+
+        args.append('-DCMAKE_CXX_STANDARD={0}'.format(
+            spec.variants['cxxstd'].value))
+        # Require standard at configure time to guarantee the
+        # compiler supports the selected standard.
+        args.append('-DCMAKE_CXX_STANDARD_REQUIRED=ON')
+
+        args.append('-DBUILD_SHARED_LIBS={0}'.format(
+            'ON' if spec.variants['shared'].value else 'OFF'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/date/package.py]: We usually use `'+shared' in spec` for boolean variants, but I don't think this is a requirement. The CMakePackage base class also has some tools to automate this, although I haven't personally used them.

Comment (gmgunter) [reply var/spack/repos/builtin/packages/date/package.py]: OK, thanks for the heads up. I agree `'+shared' in spec` is nicer syntax. 

I mostly constructed this PR by reviewing similar existing recipes and hadn't really looked over CMakePackage in much detail, TBH. So I was surprised to learn that it has such nice tools for defining these cmake cli arguments. Seems like they should be more widely used.

I'll revise it to use the base class's methods. Are you aware of any exemplary cmake-based packages that I can use as reference in the future?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/date/package.py]: `dealii` is a pretty complex CMakePackage that was recently overhauled to use those nice base class features.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**